### PR TITLE
Update some basic info on processing tx fee

### LINKED ISSUE #4: Update tx fee related default information
related to https://github.com/amolabs/amoabci/issues/148

### PR DESCRIPTION
**Pull request by h0n9**
resolves https://github.com/amolabs/docs/issues/4

### CODE REVIEW

**Comment by h0n9** [comment]
It is required to consider **WHEN** to transfer `fee` to a validator account.
while processing `DeliverTx` or `Commit`

File: protocol.md [hunk 1]
```diff
@@ -32,6 +32,7 @@ A transaction is represented by a JSON document which has the following format:
     "sender": "_sender_address_",
     "nonce": "_HEX-encoded_nonce_bytes_",
     "params": "_HEX-encoded_JSON_object_",
+	"fee": "_currency_",
```
Comment (Lbird) [inline protocol.md]: Tab? or space?

File: protocol.md [hunk 2]
```diff
@@ -53,7 +54,7 @@ A transaction is represented by a JSON document which has the following format:
 	- `"cancel"`
 	- `"revoke"`
 
-`_sender_address_` identifies the sender or originator of this transaction. `"nonce"` is a ***random*** byte sequence with the length of 20 bytes represented by HEX-encoding(See [Replay Attack](#replay-attack)). `"params"` is a HEX-encoded JSON object, which is different for each transaction type.
+`_sender_address_` identifies the sender or originator of this transaction. `"nonce"` is a ***random*** byte sequence with the length of 20 bytes represented by HEX-encoding(See [Replay Attack](#replay-attack)). `"params"` is a HEX-encoded JSON object, which is different for each transaction type. `"fee"` represents a specific amount of money expected to get transferred to a block validator after the transaction is committed to a block.
```
Comment (Lbird) [inline protocol.md]: block validator -> block proposer

File: protocol.md [hunk 3]
```diff
@@ -233,38 +234,42 @@ Any type of key-value data supposed to get stored on `stateDB` are put into the
...
+Upon receiving a `transfer` transaction from an account, an AMO blockchain node performs a validity check and transfers coins from sender's balance to recipient's balance when the transaction is valid.
 
 **Validity check:**
-1. `sender_balance` &ge; `amount`.
+1. `account_balance` &ge; `fee` + `amount`.
 
 **State change:**
-1. `sender_balance` &larr; `sender_balance` - `amount`
+1. `account_balance` &larr; `account_balance` - `fee` - `amount`
+1. `validator_balance` &larr; `validator_balance` + `fee`
```
Comment (Lbird) [inline protocol.md]: validator -> block proposer

File: protocol.md [hunk 4]
```diff
@@ -233,38 +234,42 @@ Any type of key-value data supposed to get stored on `stateDB` are put into the
...
 
 **Validity check:**
-1. `balance` &ge; `amount`
+1. `account_balance` &ge; `fee` + `amount`
 1. There is no other stake holder with the same `validator key` as this transaction.
 
 **State change:**
-1. `balance` &larr; `balance` - `amount`
+1. `account_balance` &larr; `account_balance` - `fee` - `amount`
+1. `validator_balance` &larr; `validator_balance` + `fee`
```
Comment (Lbird) [inline protocol.md]: validator -> block proposer

File: protocol.md [hunk 5]
```diff
@@ -233,38 +234,42 @@ Any type of key-value data supposed to get stored on `stateDB` are put into the
...
 1. `stake` &ge; `amount`
 1. `stake` &gt; `amount` if this account is a delegatee for any of delegated stakes
 
 **TODO:** minimum required stake to be a delegatee
 
 **State change:**
-1. `balance` &larr; `balance` + `amount`
 1. `stake` &larr; `stake` - `amount`
+1. `account_balance` &larr; `account_balance` - `fee` + `amount`
+1. `validator_balance` &larr; `validator_balance` + `fee`
```
Comment (Lbird) [inline protocol.md]: validator -> block proposer

File: protocol.md [hunk 6]
```diff
@@ -274,22 +279,25 @@ There may be users who have the intention to participate in the block production
...
 **Validity check:**
-1. `balance` &ge; `amount`
+1. `account_balance` &ge; `fee` + `amount`
 1. `to` address already has a positive stake in `stake` store
 1. the account has no previous delegatee or `to` is the same as the previous delegatee
 
 **State change:**
-1. `balance` &larr; `balance` - `amount`
+1. `account_balance` &larr; `account_balance` - `fee` - `amount`
+1. `validator_balance` &larr; `validator_balance` + `fee`
```
Comment (Lbird) [inline protocol.md]: validator -> block proposer

File: protocol.md [hunk 7]
```diff
@@ -274,22 +279,25 @@ There may be users who have the intention to participate in the block production
...
 **Validity check:**
+1. `account_balance` &ge; `fee`
 1. `delegated_stake` &ge; `amount`
 
 **State change:**
-1. `balance` &larr; `balance` + `amount`
-1. `delegated_stake` &larr; `delegated_stake` - `amount`
+1. `delegatedstake` &larr; `delegated_stake` - `amount`
+1. `account_balance` &larr; `account_balance` - `fee` + `amount`
+1. `validator_balance` &larr; `validator_balance` + `fee`
```
Comment (Lbird) [inline protocol.md]: validator -> block proposer

**Review by Lbird — CHANGES_REQUESTED** [review]

**Review by Lbird — APPROVED** [review]

**Pull request merged**